docs/src/reference/asciidoc/core/kerberos.adoc

```diff
@@ -0,0 +1,481 @@
+credentials. In the case of using an integration that does not depend on Hadoop's runtime (e.g. Storm), additional
+steps may be required to ensure that a running process has {krb} credentials available for authentication. It is
+recommended that you consult the documentation of each framework that you are using on how to configure security.
+
+[[kerberos-settings]]
+=== Setting up your environment
+
+NOTE: This documentation assumes that you have already provisioned a secure Hadoop cluster. The general process of
+deploying {krb} and securing Hadoop is beyond the scope of this documentation.
+
+Before starting, you will need to ensure that principals for your users are provisioned in your {krb} deployment,
+as well as service principals for each {es} node. To enable {krb} authentication on {es}, it must be
+https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-kerberos-realm.html[configured with a
+{krb} realm]. It is recommended that you familiarize yourself with how to configure {es} {krb} realms so that you
+can make appropriate adjustments to fit your deployment. You can find more information on how they work in the
+https://www.elastic.co/guide/en/elastic-stack-overview/current/kerberos-realm.html[Elastic Stack documentation].
+
+Additionally, you will need to https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html[
+configure the API Key Realm] in {es}. Hadoop and other distributed data processing frameworks only authenticate with
+{krb} in the process that launches a job. Once a job has been launched, the worker processes are often cut off from
+the original {krb} credentials and need some other form of authentication. Hadoop services often provide mechanisms for
+obtaining _Delegation Tokens_ during job submission. These tokens are then distributed to worker processes which use
+the tokens to authenticate on behalf of the user running the job. {ehtm} obtains API Keys in order to provide tokens
+for worker processes to authenticate with.
+
+[[kerberos-settings-eshadoop]]
+==== Connector Settings
+
+The following settings are used to configure {eh} to use {krb} authentication:
+
+`es.security.authentication` (default `simple`, or `basic` if `es.net.http.auth.user` is set)::
+Required. Similar to most Hadoop integrations, this property signals which method to use in order to authenticate with
+{es}. By default, the value is `simple`, unless `es.net.http.auth.user` is set, in which case it will default to
+`basic`. The available options for this setting are `simple` for no authentication, `basic` for basic http
+authentication, `pki` if relying on certificates, and `kerberos` if {krb} authentication over SPNEGO should be used.
+
+`es.net.spnego.auth.elasticsearch.principal` (default none)::
+Required. Details the name of the service principal that the Elasticsearch server is running as. This will usually be
```

Comment:
this is only required if the above authentication setting above is set to Kerberos, right?